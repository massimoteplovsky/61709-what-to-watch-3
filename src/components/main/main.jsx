import React from "react";
import {PropValidator} from "../../prop-validator/prop-validator";
import {connect} from 'react-redux';
import Header from '../header/header.jsx';
import MoviePromo from '../movie-promo/movie-promo.jsx';
import GenreList from '../genre-list/genre-list.jsx';
import MovieList from "../movie-list/movie-list.jsx";
import ShowMore from '../show-more/show-more.jsx';
import Footer from '../footer/footer.jsx';
import withActiveItem from '../../hocs/with-active-item/with-active-item';

const WrappedGenreList = withActiveItem(GenreList);

const Main = ({
  filteredFilms,
  onTitleClick,
  promoFilmInfo: {
    title,
    genre,
    year
  }
}) => {

  return (
    <>
      <section className="movie-card">
        <div className="movie-card__bg">
          <img src="img/bg-the-grand-budapest-hotel.jpg" alt="The Grand Budapest Hotel" />
        </div>

        <h1 className="visually-hidden">WTW</h1>

        <Header/>

        <MoviePromo
          title={title}
          genre={genre}
          year={year}
        />
      </section>

      <div className="page-content">
        <section className="catalog">
          <h2 className="catalog__title visually-hidden">Catalog</h2>

          <WrappedGenreList/>

          <MovieList
            films={filteredFilms}
            onTitleClick={onTitleClick}
          />

          <ShowMore/>
        </section>

        <Footer/>
      </div>
    </>
  );
};

Main.propTypes = {
  promoFilmInfo: PropValidator.PROMO_FILM_INFO,
  filteredFilms: PropValidator.FILMS,
  onTitleClick: PropValidator.TITLE_CLICK,
  filmCounter: PropValidator.FILM_COUNTER
};

const mapStateToProps = ({filteredFilms, filmCounter}) => ({
  filteredFilms: filteredFilms.length > filmCounter ? filteredFilms.slice(0, filmCounter) : filteredFilms,
  filmCounter
});

export {Main};

export default connect(mapStateToProps)(Main);

