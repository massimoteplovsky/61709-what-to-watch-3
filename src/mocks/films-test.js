export const films = [
  {
    id: 1,
    title: `Terminator 2. Judgment day`,
    genre: [`Action`],
    year: 1991,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: []
  },
  {
    id: 2,
    title: `1+1`,
    genre: [`Drama`, `Comedy`],
    year: 2012,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: []
  },
  {
    id: 3,
    title: `Joker`,
    genre: [`Drama`],
    year: 2019,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: []
  },
  {
    id: 4,
    title: `Once upon a time in Hollywood`,
    genre: [`Drama`],
    year: 2019,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: []
  },
  {
    id: 5,
    title: `Ford vs Ferrari`,
    genre: [`Drama`, `History`],
    year: 2019,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: []
  },
  {
    id: 6,
    title: `1917`,
    genre: [`Drama`, `History`],
    year: 2019,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: []
  },
  {
    id: 7,
    title: `Parasite`,
    genre: [`Drama`, `Comedy`],
    year: 2019,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: []
  },
  {
    id: 8,
    title: `Irishman`,
    genre: [`drama`, `criminal`],
    year: 2019,
    promoPoster: ``,
    poster: ``,
    cover: ``,
    description: ``,
    rating: 8.5,
    votes: 12000,
    director: ``,
    starring: []
  }
];

export const film = {
  id: 1,
  title: `Terminator 2. Judgment day`,
  genre: [`Action`],
  year: 1991,
  promoPoster: ``,
  poster: ``,
  cover: ``,
  description: ``,
  rating: 8.5,
  votes: 12000,
  director: ``,
  starring: []
};

export const promoFilmInfo = {
  title: `The Grand Budapest Hotel`,
  genre: `Drama`,
  year: 2014
};
