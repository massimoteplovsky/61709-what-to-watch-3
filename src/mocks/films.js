export default [
  {
    id: 1,
    title: `Terminator 2. Judgment day`,
    poster: `http://i4.ytimg.com/vi/HgV7-MJwUBw/mqdefault.jpg`
  },
  {
    id: 2,
    title: `1+1`,
    poster: `https://i.pinimg.com/736x/76/44/c0/7644c07844dd4cb28d2c289bfa6e0dec--the-intouchables-comedy-movies.jpg`
  },
  {
    id: 3,
    title: `Joker`,
    poster: `https://all.culture.ru/uploads/9b743d135092e110a8e55a3dc3ad5bc4_w375_h250.jpeg`
  },
  {
    id: 4,
    title: `Once upon a time in Hollywood`,
    poster: `https://avatars.mds.yandex.net/get-pdb/1945878/3281b62e-4539-4eb7-a26b-fa68f6572bea/s375`
  },
  {
    id: 5,
    title: `Ford vs Ferrari`,
    poster: `https://sun9-28.userapi.com/c206816/v206816832/1771c/3p_wj8sfMz8.jpg`
  },
  {
    id: 6,
    title: `1917`,
    poster: `https://sun3-13.userapi.com/c857724/v857724955/14e882/u5CFMvbvX3I.jpg`
  },
  {
    id: 7,
    title: `Parasite`,
    poster: `http://m.kino-teatr.ru/art/5395/pv_75950.jpg`
  },
  {
    id: 8,
    title: `Irishman`,
    poster: `https://sun9-31.userapi.com/c850608/v850608676/40eb4/ojB5App6exk.jpg`
  },
];
