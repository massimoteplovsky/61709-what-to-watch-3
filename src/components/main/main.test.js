import React from "react";
import renderer from "react-test-renderer";
import {films, film} from "../../mocks/films-test";
import {Main} from "./main.jsx";
import {Provider} from "react-redux";
import {Router} from "react-router-dom";
import history from "../../history";
import configureMockStore from "redux-mock-store";

const mockStore = configureMockStore([]);
let store = mockStore({
  films: {
    filteredFilms: films,
    favoriteFilms: films,
    films,
    promoFilm: null,
    filmCounter: 8,
    actualGenre: ``,
    reviews: []
  },
  user: {
    authorizationStatus: `NO_AUTH`,
    userInfo: null
  }
});

it(`<Main /> component renders correctly`, () => {
  const tree = renderer
    .create(
        <Provider store={store}>
          <Router history={history}>
            <Main
              filteredFilms={films}
              promoFilm={film}
              onChangeFilmGenre={() => {}}
            />
          </Router>
        </Provider>, {
          createNodeMock: () => {
            return {};
          }}).toJSON();
  expect(tree).toMatchSnapshot();
});

