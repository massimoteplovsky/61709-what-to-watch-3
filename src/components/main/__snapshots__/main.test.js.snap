// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> component renders correctly 1`] = `
Array [
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="The Grand Budapest Hotel"
        src="img/bg-the-grand-budapest-hotel.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href="main.html"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <div
          className="user-block__avatar"
        >
          <img
            alt="User avatar"
            height="63"
            src="img/avatar.jpg"
            width="63"
          />
        </div>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__desc"
      >
        <h2
          className="movie-card__title"
        >
          The Grand Budapest Hotel
        </h2>
        <p
          className="movie-card__meta"
        >
          <span
            className="movie-card__genre"
          >
            Drama
          </span>
          <span
            className="movie-card__year"
          >
            2014
          </span>
        </p>
        <div
          className="movie-card__buttons"
        >
          <button
            className="btn btn--play movie-card__button"
            type="button"
          >
            <svg
              height="19"
              viewBox="0 0 19 19"
              width="19"
            >
              <use
                xlinkHref="#play-s"
              />
            </svg>
            <span>
              Play
            </span>
          </button>
          <button
            className="btn btn--list movie-card__button"
            type="button"
          >
            <svg
              height="20"
              viewBox="0 0 19 20"
              width="19"
            >
              <use
                xlinkHref="#add"
              />
            </svg>
            <span>
              My list
            </span>
          </button>
          <a
            className="btn movie-card__button"
            href="add-review.html"
          >
            Add review
          </a>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item catalog__genres-item--active"
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            All genres
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Action
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Drama
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
            onClick={[Function]}
          >
            <video
              height="175"
              style={
                Object {
                  "objectFit": "cover",
                }
              }
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
            onClick={[Function]}
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Terminator 2. Judgment day
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
            onClick={[Function]}
          >
            <video
              height="175"
              style={
                Object {
                  "objectFit": "cover",
                }
              }
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
            onClick={[Function]}
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              1+1
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
            onClick={[Function]}
          >
            <video
              height="175"
              style={
                Object {
                  "objectFit": "cover",
                }
              }
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
            onClick={[Function]}
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Joker
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
            onClick={[Function]}
          >
            <video
              height="175"
              style={
                Object {
                  "objectFit": "cover",
                }
              }
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
            onClick={[Function]}
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Once upon a time in Hollywood
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
            onClick={[Function]}
          >
            <video
              height="175"
              style={
                Object {
                  "objectFit": "cover",
                }
              }
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
            onClick={[Function]}
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Ford vs Ferrari
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
            onClick={[Function]}
          >
            <video
              height="175"
              style={
                Object {
                  "objectFit": "cover",
                }
              }
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
            onClick={[Function]}
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              1917
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
            onClick={[Function]}
          >
            <video
              height="175"
              style={
                Object {
                  "objectFit": "cover",
                }
              }
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
            onClick={[Function]}
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Parasite
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
            onClick={[Function]}
          >
            <video
              height="175"
              style={
                Object {
                  "objectFit": "cover",
                }
              }
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
            onClick={[Function]}
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Irishman
            </a>
          </h3>
        </article>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="main.html"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          © 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
